File: app/src/main/java/com/example/toothfairy/util/Event.java
package com.example.toothfairy.util;

public class Event<T> {
    private boolean hasBeenHandled = false;
    private T content;

    public Event(T content) { this.content = content; }

    public T getContentIfNotHandled(){
        if(hasBeenHandled) return null;

        hasBeenHandled = true;
        return content;
    }

    public void setContent(T content){ this.content = content; }
}

File: app/src/main/java/com/example/toothfairy/util/Event.kt
package com.example.toothfairy.util

/** 중복으로 이벤트가 발생하는 것을 막기위한 클래스 */
class Event<T>(private var content: T) {
    private var hasBeenHandled = false

    fun contentIfNotHandled(): T?{
        if(hasBeenHandled) return null

        hasBeenHandled = true
        return content
    }

    fun setContent(content: T) {
        this.content = content
    }
}